src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -68,25 +60,24 @@ public static IServiceCollection AddApplicationInsightsTelemetry(this IServiceCo
             services.AddSingleton<ITelemetryModule, DependencyTrackingTelemetryModule>();
 #endif
 
-            services.AddSingleton<TelemetryConfiguration>(serviceProvider =>
+            services.AddOptions();
+            services.AddSingleton<IOptions<TelemetryConfiguration>, TelemetryConfigurationOptions>();
+            services.AddSingleton<IConfigureOptions<TelemetryConfiguration>, TelemetryConfigurationOptionsSetup>();
+            if (options != null)
             {
-                var telemetryConfiguration = TelemetryConfiguration.Active;
-                AddTelemetryChannelAndProcessorsForFullFramework(serviceProvider, telemetryConfiguration, options);
-                telemetryConfiguration.TelemetryChannel = serviceProvider.GetService<ITelemetryChannel>() ?? telemetryConfiguration.TelemetryChannel;
-                AddTelemetryConfiguration(config, telemetryConfiguration);
-                AddServicesToCollection(serviceProvider, telemetryConfiguration.TelemetryInitializers);
-                InitializeModulesWithConfiguration(serviceProvider, telemetryConfiguration);
-                return telemetryConfiguration;
-            });
+                services.Configure<ApplicationInsightsServiceOptions>(options);
+            }
+            services.AddSingleton<TelemetryConfiguration>(provider => provider.GetService<IOptions<TelemetryConfiguration>>().Value);
 
             services.AddSingleton<TelemetryClient>();
 
-            services.AddScoped<RequestTelemetry>((svcs) => {
-                // Default constructor need to be used
-                var rt = new RequestTelemetry();
-                return rt;
-            });
-            
+            services.AddSingleton<ApplicationInsightsInitializer, ApplicationInsightsInitializer>();
+            services.AddSingleton<IApplicationInsightDiagnosticListener, AspNetCoreHostingDiagnosticListener>();
+            services.AddSingleton<IApplicationInsightDiagnosticListener, AspNetCoreMvcDiagnosticsListener>();
+            services.AddSingleton<IStartupFilter, ApplicationInsightsStartupFilter>();
```

Comment:
AspNetCoreHostingDiagnosticListener injects `TelemetryClient` that injects `TelemetryConfiguration` that requires `IOptions` infrastructure to run and initialize.